From 8c5dbf54d810c7d91800ee754b59f1640a877544 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Fri, 23 Jan 2026 05:43:19 +0100
Subject: [PATCH 1/1] Fix environment variables

Set the XDG_CURRENT_DESKTOP environment variable for gsettings overrides
per session. Import the DISPLAY environment variable into the systemd
manager environment block to show nm-applet notifications and to fix
the keyboard layout to correspond with the indicator-keyboard icon.  
Import the XDG_SESSION_CLASS environment variable, which is used
by indicator-sound.

Signed-off-by: c4pp4
---
 src/unity-greeter.vala | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/unity-greeter.vala b/src/unity-greeter.vala
index 929fa2c..a5db5f7 100644
--- a/src/unity-greeter.vala
+++ b/src/unity-greeter.vala
@@ -540,6 +540,16 @@ public class UnityGreeter
 
     public static int main (string[] args)
     {
+        try
+        {
+            Process.spawn_command_line_sync ("systemctl --user import-environment DISPLAY XDG_SESSION_CLASS");
+        }
+        catch (Error e)
+        {
+            warning ("Error importing environment variables: %s", e.message);
+        }
+
+        Environment.set_variable ("XDG_CURRENT_DESKTOP", "unity-greeter", false);
 
         /* Disable the stupid global menubar */
         Environment.unset_variable ("UBUNTU_MENUPROXY");
-- 
2.52.0

